packages/shared/src/WithNativeNavigation.js

```diff
@@ -18,16 +20,31 @@ function withNativeNavigation<Props: {}>(
 ): React.ComponentType<$Diff<Props, InjectedProps>> {
   return class WithNativeNavigation extends React.Component<*> {
     currentIndex: number;
+    navigator: NavigatorType;
 
     constructor(props) {
       super(props);
       this.currentIndex = 0;
     }
+
+    nestings = function nestings(nav) {
+      const nestedNav =
+        nav.routes && nav.index !== undefined && nav.routes[nav.index];
+      if (!nestedNav) {
+        return nav.params && nav.params.isModal;
+      }
+      return nestings(nestedNav);
+    };
+
     onNavigationStateChange = (
       previousState: NavigationState,
       currentState: NavigationState,
     ) => {
-      if (currentState.index === 0) {
+      const nav = this.navigator.state.nav;
+
+      if (this.nestings(nav)) {
+        GestureController.disableGestures(moduleName);
```

Comment:
Just so I understand, this line is the magic that disables the swipe in the modal?

Reply:
Exactly! It disabling native gestures from the js side.